cli/src/debugger.rs

```diff
@@ -1,18 +1,35 @@
 use crate::common::CliError;
 
 use capstone::Capstone;
+use num_traits::Num;
 use probe_rs::architecture::arm::CortexDump;
 use probe_rs::debug::DebugInfo;
 use probe_rs::{Core, CoreRegisterAddress, MemoryInterface};
+
 use std::fs::File;
 use std::{io::prelude::*, time::Duration};
 
+use parse_int::parse;
+
 pub struct DebugCli {
     commands: Vec<Command>,
 }
 
-fn parse_u32_hex(s: &str) -> u32 {
-    u32::from_str_radix(s, 16).expect("Couldn't parse hex number")
+/// Parse the argument at the given index.
+fn get_int_argument<T: Num>(args: &[&str], index: usize) -> Result<T, CliError>
+where
+    <T as Num>::FromStrRadixErr: std::error::Error,
+    <T as Num>::FromStrRadixErr: Send,
+    <T as Num>::FromStrRadixErr: Sync,
+    <T as Num>::FromStrRadixErr: 'static,
+{
+    let arg_str = args.get(index).ok_or(CliError::MissingArgument)?;
+
+    parse::<T>(arg_str).map_err(|e| CliError::ArgumentParseError {
+        argument_index: 0,
```

Comment:
I think this should be `index` and not `0`.

Reply:
Good catch!